imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,197 @@
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.decom import decom_packets
+from imap_processing.ialirt.l0.decom_ialirt import generate_xarray
+
+IALIRT_PACKET_LENGTH = 1464
+
+
+@pytest.fixture(scope="session")
+def xtce_ialirt_path():
+    """Returns the xtce auxiliary directory."""
+    return imap_module_directory / "ialirt" / "packet_definitions" / "ialirt.xml"
+
+
+@pytest.fixture(scope="session")
+def binary_packet_path():
+    """
+    Creates a binary file from the text packet data, which is more representative
+    of the actual operational environment. The binary file is deleted after the
+    test session.
+    """
+    packet_path = (
+        imap_module_directory
+        / "ialirt"
+        / "tests"
+        / "test_data"
+        / "l0"
+        / "IALiRT Raw Packet Telemetry.txt"
+    )
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
```

Comment:
We want to delete the temporary file on clean-up, so we don't want this to be `delete=False`.

https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html
I think you can get what you are after here with the `tmp_path` fixture from pytest directly, and it will be cleaned up for you after the yield is finished. (meaning you don't need the unlink later either.